### PR TITLE
add routes for group and privat chats

### PR DESCRIPTION
**Pull request by ReginaVarybrus**
In this pr I added routes for group and privat chats. Also I moved the MessageBar component outside the parent component,  changed some styles for upload file button and replace icons in the MessageBar component with react icons library.

### CODE REVIEW

File: src/components/DMsSearchBar/DMsSearchBar.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { useState } from 'react';
```
Comment (DimaShubin) [inline src/components/DMsSearchBar/DMsSearchBar.jsx]: We already have similar component. Take a look on RoomSearchBar component.

File: src/routes/DmsChatRoute/DmsChatRoute.jsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import DMsSearchBar from '@/components/DMsSearchBar/DMsSearchBar';
+import Chat from '@/components/Chat/Chat';
+import { ChatRouteStyled } from '../RoomsChatRoute/ChatRouteStyled.js';
+
+const DmsChatRoute = () => (
+  <ChatRouteStyled>
+    <DMsSearchBar />
```
Comment (DimaShubin) [inline src/routes/DmsChatRoute/DmsChatRoute.jsx]: Do we really need counties search input in DM route?

File: src/components/DMsSearchBar/DMsSearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import DMsList from '../DMsList/DMsList';
+import ChatMap from '../ChatMap/ChatMap';
+
+const DMsSearchBar = () => {
+  const [openMap, setOpenMap] = useState(false);
+  const handleOpen = () => setOpenMap(true);
+  const handleClose = () => setOpenMap(false);
+
+  return (
+    <SearchBarStyled>
```
Comment (DimaShubin) [inline src/components/DMsSearchBar/DMsSearchBar.jsx]: I suppose you could reuse already existing component.

File: src/components/SearchInput/SearchInput.jsx [hunk 4]
```diff
@@ -73,20 +76,20 @@ const SearchInput = () => {
...
-      </IconSearch>
-
+      <IconSearch />
       {showItem && (
         <ListWrapper>
           <ListItems>
             <ScrollBar>
-              {filterCountries === 0 ? (
-                <p>BlaBla</p>
+              {filterCountries.length === 0 ? (
```
Comment (DimaShubin) [inline src/components/SearchInput/SearchInput.jsx]: filterCountries.length ? (

File: src/components/SideBar/SideBarStyled.js [hunk 5]
```diff
@@ -1,61 +1,88 @@
...
   flex-direction: column;
   justify-content: space-between;
-  align-items: center;
-  width: 56px;
-  height: 76px;
-  margin-top: 48px;
-  cursor: pointer;
+  height: 100%;
+  margin: 48px 0;
+  margin: 48px 0;
```
Comment (DimaShubin) [inline src/components/SideBar/SideBarStyled.js]: duplication

File: src/components/SideBar/SideBarStyled.js [hunk 6]
```diff
@@ -1,61 +1,88 @@
...
+`;
+
+export const Text = styled.p`
+  line-height: 20px;
+  letter-spacing: -0.011em;
+  color: var(--color-grey-9);
+`;
+
+export const UserIcon = styled(TbUser)`
+  width: 24px;
```
Comment (DimaShubin) [inline src/components/SideBar/SideBarStyled.js]: Seems like all icons below uses the same styles. We should make a bit clear and move these styles to a variable and use this variable for the rest of icons.

File: src/components/SideBar/SideBarStyled.js [hunk 7]
```diff
@@ -1,61 +1,88 @@
...
+  width: 56px;
+  height: 76px;
+  cursor: pointer;
+
+  &:hover {
+    ${Text} {
+      color: var(--color-blue-3);
+    }
+    ${DMsIcon}, ${RoomsIcon}, ${UserIcon}, ${LogoutIcon} {
+      stroke: var(--color-blue-3);
```
Comment (DimaShubin) [inline src/components/SideBar/SideBarStyled.js]: Try to use stroke: currentColor; instead.

File: src/components/SideBar/SideBarStyled.js [hunk 8]
```diff
@@ -1,61 +1,88 @@
...
+    }
+  }
+
+  ${Text} {
+    color: ${({ $isActive }) =>
+      $isActive ? 'var(--color-brand-blue)' : 'var(--color-grey-9)'};
+    font-weight: ${({ $isActive }) => ($isActive ? '700' : '400')};
+  }
+  ${DMsIcon}, ${RoomsIcon}, ${UserIcon} {
+    stroke: ${({ $isActive }) =>
```
Comment (DimaShubin) [inline src/components/SideBar/SideBarStyled.js]: currentColor

File: vite.config.js [hunk 9]
```diff
@@ -24,6 +24,14 @@ export default defineConfig({
     }),
     viteTsconfigPaths(),
   ],
+  optimizeDeps: {
```
Comment (DimaShubin) [inline vite.config.js]: What is it for?

File: vite.config.js [hunk 10]
```diff
@@ -24,6 +24,14 @@ export default defineConfig({
...
+    include: ['linked-dep'],
+  },
+  build: {
+    commonjsOptions: {
+      include: [/linked-dep/, /node_modules/],
```
Comment (DimaShubin) [inline vite.config.js]: Do we have a folder /linked-dep/ ? Did I miss smth?

**Comment by DimaShubin** [comment]
Try to resolve conflicts.

File: dist/vite.config.dev.js [hunk 11]
```diff
@@ -0,0 +1,49 @@
+"use strict";
```
Comment (DimaShubin) [inline dist/vite.config.dev.js]: Delete it.

File: src/components/SearchInput/SearchInput.jsx (hunk 4 above)
Comment (DimaShubin) [reply src/components/SearchInput/SearchInput.jsx]:
```suggestion
              {filterCountries.length ? (
```

File: src/components/ChatFirstLoading/ChatFirstLoading.jsx [hunk 12]
```diff
@@ -0,0 +1,12 @@
+import { ChatStyled, LogoBox, Logo } from './ChatFirstLoadingStyled.js';
+
```
Comment (DimaShubin) [inline src/components/ChatFirstLoading/ChatFirstLoading.jsx:3]:
```suggestion
import logo from '../../../public/logo.svg';
```

File: src/components/ChatFirstLoading/ChatFirstLoading.jsx [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+const ChatFistLoading = () => (
+  <ChatStyled>
+    <LogoBox>
+      <Logo src="../../../public/logo.svg" alt="logo" />
```
Comment (DimaShubin) [inline src/components/ChatFirstLoading/ChatFirstLoading.jsx]:
```suggestion
      <Logo src={logo} alt="logo" />
```

File: src/components/ChatFirstLoading/ChatFirstLoadingStyled.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+import styled from 'styled-components';
+
+export const ChatStyled = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  width: 100%;
+  min-height: 100vh;
```
Comment (DimaShubin) [inline src/components/ChatFirstLoading/ChatFirstLoadingStyled.js]: This rule will afect the whole page height and add vertical scrol

File: src/components/DMsList/DMsList.jsx [hunk 15]
```diff
@@ -1,45 +1,39 @@
...
-    <Wrapper>
-      {chats.length === 0 && (
+    <ListStyled>
+      {chats?.length ? (
+        <ListItems>
+          <SimpleBar style={{ maxHeight: 570 }}>
+            {chats.map(name => (
+              <Item key={name}>
+                <Avatar>
+                  <img src="../../../public/img/Avatar.png" alt="avatar" />
```
Comment (DimaShubin) [inline src/components/DMsList/DMsList.jsx]: Use file import instead. 

import avatarImage from '../../../public/img/Avatar.png';


```suggestion
                  <img src={avatarImage} alt="avatar" />
```

File: src/components/Layout/LayoutStyled.js [hunk 16]
```diff
@@ -1,7 +1,14 @@
 import styled from 'styled-components';
+import { device } from '@/constants/mediaQueries';
 
 export const LayoutStyled = styled.div`
   display: grid;
-  grid-template-columns: 120px 1fr;
+  grid-template-columns: 90px 1fr;
   background-color: var(--outlet-background-color);
+  @media ${device.tablet} {
+    grid-template-columns: 90px 1fr;
```
Comment (DimaShubin) [inline src/components/Layout/LayoutStyled.js]: I beliave this is redundant.

File: src/components/RoomsList/RoomsList.jsx [hunk 17]
```diff
@@ -1,5 +1,5 @@
 import {
```
Comment (DimaShubin) [inline src/components/RoomsList/RoomsList.jsx:1]: Seems like we don't need this file anymore.

File: src/components/SearchBar/SearchBar.jsx [hunk 18]
```diff
@@ -3,23 +3,22 @@ import Backdrop from '@mui/material/Backdrop';
...
 import Fade from '@mui/material/Fade';
 
-import PropTypes from 'prop-types';
-import { Wrapper, ButtonMapOpen, MapBox } from './SearchBarStyled';
+import { Outlet } from 'react-router-dom';
+import { SearchBarStyled, ButtonMapOpen, MapBox } from './SearchBarStyled';
 import SearchInput from '../SearchInput/SearchInput';
-import DMsList from '../DMsList/DMsList';
-import RoomsList from '../RoomsList/RoomsList';
+// import RoomsList from '../RoomsList/RoomsList';
```
Comment (DimaShubin) [inline src/components/SearchBar/SearchBar.jsx]: ???

File: package.json [hunk 19]
```diff
@@ -14,12 +14,14 @@
     "@vitejs/plugin-react": "^4.2.1",
     "axios": "^1.5.1",
     "babel-plugin-styled-components": "^2.1.4",
+    "esbuild": "^0.20.1",
```
Comment (DimaShubin) [inline package.json]: Why do you need this?

File: vite.config.js [hunk 20]
```diff
@@ -34,5 +39,6 @@ export default defineConfig({
     alias: {
       '@': path.resolve(__dirname, './src'),
     },
+    preserveSymlinks: true,
```
Comment (DimaShubin) [inline vite.config.js]: What is it for?

File: src/components/ChatFirstLoading/ChatFirstLoadingStyled.js (hunk 14 above)
Comment (ReginaVarybrus) [reply src/components/ChatFirstLoading/ChatFirstLoadingStyled.js]: I'll chage it when I'll started to add styles to this page.

File: src/components/Layout/LayoutStyled.js (hunk 16 above)
Comment (ReginaVarybrus) [reply src/components/Layout/LayoutStyled.js]: It is from UI/UX we had conversation and he said that for this mediaQueries should be 90px width on SideBar component

File: package.json (hunk 19 above)
Comment (ReginaVarybrus) [reply package.json]: Deleted

File: vite.config.js (hunk 20 above)
Comment (ReginaVarybrus) [reply vite.config.js]: Deleted

File: src/components/RegisterForm/RegisterForm.jsx [hunk 21]
```diff
@@ -1,6 +1,7 @@
 import { useFormik } from 'formik';
 import * as yup from 'yup';
 import { register } from '@/redux-store/AuthOperations/AuthOperations';
+import { useNavigate } from 'react-router-dom';
```
Comment (DimaShubin) [inline src/components/RegisterForm/RegisterForm.jsx:3]: It has already been defined below.

File: public/img/Avatar.png
Comment (DimaShubin) [inline public/img/Avatar.png:1]: Why did you moved it to the public directory?
image

File: src/components/MessageBar/MessageBar.jsx [hunk 22]
```diff
@@ -0,0 +1,46 @@
...
+        </ButtonAttachFile>
+        <TextareaAutosize
+          aria-label="empty textarea"
+          placeholder="Type here"
+          value={value}
+          onChange={handleChange}
+          ref={textAreaRef}
+          maxLength="1000"
+        />
+        <ButtonSendMessage isInputNotEmpty={isInputNotEmpty}>
```
Comment (DimaShubin) [inline src/components/MessageBar/MessageBar.jsx]:
```suggestion
        <ButtonSendMessage $isInputNotEmpty={isInputNotEmpty}>
```

File: src/components/DMsList/DMsList.jsx [hunk 23]
```diff
@@ -1,45 +1,40 @@
...
-  ListItems,
-  Item,
-  Avatar,
-  ChatName,
-  MessageDay,
-} from './DMsListStyled';
+import { Item, Avatar, ChatName, MessageDay } from './DMsListStyled';
+import { ListStyled, Text, ListItems } from '../RoomsList/RoomsListStyled.js';
 import 'simplebar-react/dist/simplebar.min.css';
+import avatarImage from '../../../public/img/Avatar.png';
```
Comment (DimaShubin) [inline src/components/DMsList/DMsList.jsx]: This one also haven't be imported from /public

File: src/components/RoomsList/RoomsListStyled.js [hunk 24]
```diff
@@ -1,26 +1,32 @@
...
 
 export const Text = styled.p`
   text-align: center;
   font-size: 14px;
   color: var(--color-grey-9);
-  margin: 0 16px;
 `;
 
 export const ListItems = styled.ul`
   position: absolute;
```
Comment (DimaShubin) [inline src/components/RoomsList/RoomsListStyled.js]: Why is it positioned absolutly?

File: src/components/Layout/LayoutStyled.js (hunk 16 above)
Comment (DimaShubin) [reply src/components/Layout/LayoutStyled.js]: You already defined the same styles 3 rows above.

File: vite.config.js (hunk 20 above)
Comment (DimaShubin) [reply vite.config.js]: дє????

File: src/components/SearchInput/SearchInput.jsx (hunk 4 above)
Comment (ReginaVarybrus) [reply src/components/SearchInput/SearchInput.jsx]: I finally did it)

File: public/img/Avatar.png
Comment (ReginaVarybrus) [reply public/img/Avatar.png:1]: I moved it back into the images directory

**Review by DimaShubin — APPROVED** [review]

**Pull request merged**